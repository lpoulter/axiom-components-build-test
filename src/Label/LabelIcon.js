import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Icon from '../Icon/Icon';
import Link from '../Typography/Link';
import './LabelIcon.css';

export const LabelIconRef = 'LabelIcon';

export default class LabelIcon extends Component {
  static propTypes = {
    /** SKIP */
    color: PropTypes.string,
    /** SKIP */
    isEnd: PropTypes.bool,
    /** SKIP */
    isStart: PropTypes.bool,
    /** Name of the icon. See <Icon>. */
    name: PropTypes.string.isRequired,
  };

  static typeRef = LabelIconRef;

  render() {
    const { color, isEnd, isStart, name, ...rest } = this.props;
    const classes = classnames('ax-label__icon', {
      'ax-label__icon--start': isStart,
      'ax-label__icon--end': isEnd,
    });

    const linkStyle = {
      white: 'subtle',
      success: 'night',
      error: 'night',
    }[color];

    return (
      <span className={ classes }>
        <Link { ...rest } style={ linkStyle }>
          <Icon name={ name } size="1rem" />
        </Link>
      </span>
    );
  }
}
