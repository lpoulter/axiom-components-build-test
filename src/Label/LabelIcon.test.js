import React from 'react';
import renderer from 'react-test-renderer';
import LabelIcon from './LabelIcon';

const getComponent = (props = {}) =>
  renderer.create(
    <LabelIcon { ...props } name="twitter" />
  );

describe('LabelIcon', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with end spacing', () => {
    const component = getComponent({ isEnd: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with start spacing', () => {
    const component = getComponent({ isStart: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with color', () => {
    ['white', 'success', 'error'].forEach((color) => {
      it(color, () => {
        const component = getComponent({ color });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
