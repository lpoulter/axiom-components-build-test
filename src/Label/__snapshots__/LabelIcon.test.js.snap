// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelIcon renders with color error 1`] = `
<span
  className="ax-label__icon"
>
  <a
    className="ax-link ax-link--style-night"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </a>
</span>
`;

exports[`LabelIcon renders with color success 1`] = `
<span
  className="ax-label__icon"
>
  <a
    className="ax-link ax-link--style-night"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </a>
</span>
`;

exports[`LabelIcon renders with color white 1`] = `
<span
  className="ax-label__icon"
>
  <a
    className="ax-link ax-link--style-subtle"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </a>
</span>
`;

exports[`LabelIcon renders with defaultProps 1`] = `
<span
  className="ax-label__icon"
>
  <a
    className="ax-link ax-link--style-normal"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </a>
</span>
`;

exports[`LabelIcon renders with end spacing 1`] = `
<span
  className="ax-label__icon ax-label__icon--end"
>
  <a
    className="ax-link ax-link--style-normal"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </a>
</span>
`;

exports[`LabelIcon renders with start spacing 1`] = `
<span
  className="ax-label__icon ax-label__icon--start"
>
  <a
    className="ax-link ax-link--style-normal"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </a>
</span>
`;
